src/rgw/rgw_sal_motr.cc

```diff
@@ -1034,6 +1034,19 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
     }
   }
 
+  // Sort the object versions
+  vector<rgw_bucket_dir_entry>::iterator iter;
+
+  for (iter = results.objs.begin(); iter != results.objs.end(); ++iter) {
```

Comment:
why are we iterating over `results.objs`??
Just the `sort()` function should do r8?

Reply:
yeah, lamda function would iterate through whole list anyways !?

Reply:
Yes, I had added iterator to do more debugging.
We can simply add sort() function. 
Done all changes as suggested.